### PR TITLE
Hotfix-OCR: Throw error if document is not found

### PR DESCRIPTION
**Pull request by VukManojlovic**
Skip sample / return failed function request if detected document borders fail to represent a rectangle

### CODE REVIEW

File: tasks/document-ocr-fn/resources/function/function.py [hunk 1]
```diff
@@ -22,6 +22,8 @@ def response(requestData: dict[str, Any]) -> dict[str, Any]:
 
     predictedMask = detect_document.run(segmentationModel, imagePath)
     predictedMask = processMask(predictedMask)
+    if predictedMask is None:
+        return functions.badRequest("Failed to determine document borders")
```
Comment (dule1322) [inline tasks/document-ocr-fn/resources/function/function.py:26]: Let's change message to something easily understandable like: "Failed to find document on the provided image". This is not necessarily a bad request if the provided image has a document but model failed to detect it. Is there a better HTTP status code to use for when our model did not find a document on a valid image?

**Review by dule1322 — APPROVED** [review]

File: tasks/document-ocr-fn/resources/function/function.py (hunk 1 above)
Comment (VukManojlovic) [reply tasks/document-ocr-fn/resources/function/function.py:26]: How about we use HTTP status code 404 "Not found"

Comment (dule1322) [reply tasks/document-ocr-fn/resources/function/function.py]: Maybe 500 would be enough - `Internal server error`.

File: tasks/document-ocr-fn/resources/function/image_segmentation.py [hunk 2]
```diff
@@ -16,14 +16,18 @@ def findRectangle(mask: np.ndarray) -> Optional[np.ndarray]:
...
             return
 
         contour = max(contours, key = cv2.contourArea)
 
         epsilon = 0.02 * cv2.arcLength(contour, True)
         rectangle = cv2.approxPolyDP(contour, epsilon, True)
 
+        if rectangle.shape[0] < 4:
+            logging.error("Failed to find document")
+            return
```
Comment (dule1322) [inline tasks/document-ocr-fn/resources/function/image_segmentation.py:29]: This should return `None`, right? As well as the line 20 in this function.

**Review by dule1322 — CHANGES_REQUESTED** [review]

**Review by dule1322 — APPROVED** [review]

**Pull request merged**